### ISSUE TITLE
Feature request:auto save

### ISSUE THREAD

**Issue by realdreams** [notepadqq/notepadqq#128]
Adding auto save feature to save all current files every x minutes

**Comment by leogaggl** [comment]
Would definitely second this feature request. It's what stops me from moving off Geany ....

**Comment by darealshinji** [comment]
Sounds really good. How is autosaving usually handled in other editors? Do they overwrite the original files, save copies in the original directory or in a hidden user directory?

**Comment by leogaggl** [comment]
Multiple ways I have found. Most commonly they append some extension such as .autosave, .bak or similar. Others just save in regular intervals and overwrite original. I personally prefer the automatic saving of the original rather than separate extensions. Since you have always got the REDO functions or (in case of a crash) your source control system. But at least you have your work saved. 

Since I use text editors to keep the odd notes this function is very important to me. I don't care about how it's done. I just need to be sure that stuff I paste in gets saved without me having to constantly click 'Save' or key shortcut.

**Comment by asdfljdasflkjwfd** [comment]
Being able to keep at least one old version is crucial imo. An accidental "select all" and then something else can be disastrous combined with auto save. I would not count on redo for this scenario because once the file is saved it can be closed without any prompt.

**Comment by leogaggl** [comment]
@asdfljdasflkjwfd good point. it's generally not my use-case. but I guess that is why the keeping previous version is the most common implementation. 

Just leads to lot's of hidden files & cleanups. But I'd agree to err on the safe side with this one.

**Comment by 717** [comment]
+1 
to this because this feature would be very useful to simulate a .io website, which I use to practice my programming, on a localhost. I use livereload to display file changes made to a browser from my .html/.css/etc files, and it would be very helpful to see the changes I make as I type them. Especially as I change the .sass file. 

hmmm April 3, 2015 -> 6 days ago -> 1/3/2016

**Comment by danieleds** [comment]
How should a user restore an old version?

**Comment by dagostinelli** [comment]
This is related to #266
